src/Waives.Pipelines/WaivesApi.cs

```diff
@@ -126,22 +55,25 @@ public static async Task<WaivesClient> Login(string clientId, string clientSecre
         /// }
         /// ]]>
         /// </example>
-        /// <param name="deleteExistingDocuments">If set to <c>true</c>, it will (immediately) delete all documents
-        /// in existence in the Waives account; if set to <c>false</c>, no such clean up will be completed.
-        /// Defaults to <c>true</c>.</param>
-        /// <param name="logger">A logger that will receive log messages from the pipeline and the underlying <see cref="WaivesClient"/></param>
-        /// <param name="maxConcurrency">The maximum number of documents to process concurrently.</param>
+        /// <param name="options"></param>
         /// <returns>A new <see cref="Pipeline"/> instance with which you can
         /// configure your document processing pipeline.</returns>
-        public static Pipeline CreatePipeline(bool deleteExistingDocuments = true, ILogger logger = null, int maxConcurrency = RateLimiter.DefaultMaximumConcurrentDocuments)
+        public static Pipeline CreatePipeline(WaivesOptions options)
```

Comment:
:muscle: I realise it's a breaking change, but I think this should probably be an async function to avoid having to do the `Task.Run`s.

Reply:
Given we're already making a breaking change by changing the params, I think it'd be worth it.